libbeat/template/processor.go

```diff
@@ -120,7 +120,15 @@ func (p *Processor) Process(fields mapping.Fields, state *fieldState, output com
 		}
 
 		if len(indexMapping) > 0 {
-			output.Put(mapping.GenerateKey(field.Name), indexMapping)
+			if field.DynamicTemplate {
```

Comment:
Should we check here that `ObjectTypeMappingType` & `ObjectType` were not provided? This should avoid human mistake if perhaps `dynamic_template` is misunderstood by the dev.

Reply:
I've added validation in libbeat/mapping to prevent DynamicTemplate + ObjectTypeParams/ObjectType.